java/src/edu/wpi/first/wpilibj/networktables/NetworkTable.java

```diff
@@ -850,7 +850,7 @@ public boolean putRaw(String key, ByteBuffer value, int len) {
   public boolean putValue(String key, Object value) throws IllegalArgumentException {
     if (value instanceof Boolean)
       return NetworkTablesJNI.putBoolean(path + PATH_SEPARATOR + key, ((Boolean)value).booleanValue());
-    else if (value instanceof Double)
+    else if (value instanceof Number && !(value instanceof Long))
       return NetworkTablesJNI.putDouble(path + PATH_SEPARATOR + key, ((Double)value).doubleValue());
```

Comment:
Can Number be cast to a Double? Or does this need to be cast to Double then the double taken from there?